modin/pandas/series.py

```diff
@@ -1764,9 +1764,9 @@ def _prepare_inter_op(self, other):
         """
         if isinstance(other, Series):
             new_self = self.copy()
-            new_self.name = "__reduced__"
+            new_self.name = self.name
```

Comment:
What reason is for this change?

Reply:
This is fix of additional bug in current master. Code to reproduce:
```python
import pandas
import modin.pandas as pd

pd_df = pandas.DataFrame({'a': [0,1]})
md_df = pd.DataFrame({'a': [0,1]})

pd_res = pd_df.a * pd_df.a
md_res = md_df.a * md_df.a

print(pd_res)
print(md_res)
```
Log:
```
0    0
1    1
Name: a, dtype: int64
0    0
1    1
dtype: int64
```